gateway/handlers/forwarding_proxy.go

```diff
@@ -46,7 +47,18 @@ func MakeForwardingProxyHandler(proxy *types.HTTPClientReverseProxy, notifiers [
 	}
 }
 
-func buildUpstreamRequest(r *http.Request, url string) *http.Request {
+func buildUpstreamRequest(r *http.Request, baseURL string, requestURL string) *http.Request {
+	url := baseURL
+
+	if requestURL != "" {
+		matcher, _ := regexp.Compile("/?function/([^/?]+)([^?]*)")
```

Comment:
Hey, just curious what the regex is for?

Reply:
Good question!  I allowed some of my regex related comments to fall by the wayside when I refactored.  I will add them back...

So, when building the upstream URL, when forwarding it for a function, it is helpful to strip off the `/function/{functionName}` prefix to the Gateway way.

That is, if I request:  `http://gateway:8080/function/xyz/my/path` it is nicer for it to forward to:  'http://xyzcontainer:8080/my/path` than `http://xyzcontainer:8080/function/xyz/my/path`.

This regex lifts out the URL after `/function/{functionName}` and appends it to the baseURL.

This should only be done for function forwarding though, as the Gateway also forwards other types of requests and shouldn't mangle those.